FirebaseMessaging/Sources/Token/FIRMessagingTokenManager.m

```diff
@@ -655,42 +653,51 @@ - (void)setAPNSToken:(NSData *)APNSToken withUserInfo:(NSDictionary *)userInfo {
-          FIRMessaging_STRONGIFY(self);
-          if (self == nil) {
-            FIRMessagingLoggerError(kFIRMessagingMessageCodeInstanceID017,
-                                    @"Instance ID shut down during token reset. Aborting");
-            return;
-          }
-          if (self.currentAPNSInfo == nil) {
-            FIRMessagingLoggerError(kFIRMessagingMessageCodeInstanceID018,
-                                    @"apnsTokenData was set to nil during token reset. Aborting");
-            return;
-          }
+    [self.installations installationIDWithCompletion:^(NSString *_Nullable identifier,
+                                                       NSError *_Nullable error) {
+      FIRMessaging_STRONGIFY(self);
+      if (self == nil) {
+        FIRMessagingLoggerError(kFIRMessagingMessageCodeInstanceID017,
+                                @"Instance ID shut down during token reset. Aborting");
+        return;
+      }
+      if (self.currentAPNSInfo == nil) {
+        FIRMessagingLoggerError(kFIRMessagingMessageCodeInstanceID018,
+                                @"apnsTokenData was set to nil during token reset. Aborting");
+        return;
+      }
 
-          NSMutableDictionary *tokenOptions = [@{
-            kFIRMessagingTokenOptionsAPNSKey : self.currentAPNSInfo.deviceToken,
-            kFIRMessagingTokenOptionsAPNSIsSandboxKey : @(isSandboxApp)
-          } mutableCopy];
-          if (self.firebaseAppID) {
-            tokenOptions[kFIRMessagingTokenOptionsFirebaseAppIDKey] = self.firebaseAppID;
-          }
+      NSMutableDictionary *tokenOptions = [@{
+        kFIRMessagingTokenOptionsAPNSKey : self.currentAPNSInfo.deviceToken,
+        kFIRMessagingTokenOptionsAPNSIsSandboxKey : @(isSandboxApp)
+      } mutableCopy];
+      if (self.firebaseAppID) {
+        tokenOptions[kFIRMessagingTokenOptionsFirebaseAppIDKey] = self.firebaseAppID;
+      }
 
-          for (FIRMessagingTokenInfo *tokenInfo in invalidatedTokens) {
-            [self fetchNewTokenWithAuthorizedEntity:tokenInfo.authorizedEntity
-                                              scope:tokenInfo.scope
-                                         instanceID:identifier
-                                            options:tokenOptions
-                                            handler:^(NSString *_Nullable token,
-                                                      NSError *_Nullable error){
+      for (FIRMessagingTokenInfo *tokenInfo in invalidatedTokens) {
+        [self fetchNewTokenWithAuthorizedEntity:tokenInfo.authorizedEntity
+                                          scope:tokenInfo.scope
+                                     instanceID:identifier
+                                        options:tokenOptions
+                                        handler:^(NSString *_Nullable token,
+                                                  NSError *_Nullable error){
 
```

Comment:
I recognize this isn't new, but can we maybe add a comment here mentioning why we don't need to do anything in the completion block?

Reply:
it just needs to call the right API to send a token request which has a completion handler.